src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -89,26 +89,58 @@ function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,3},
   end
 
   if (alpha_smooth)
-    # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
-    # Copy alpha values such that smoothing is indpedenent of the element access order
-    alpha_tmp .= alpha
-
-    # Loop over interfaces
-    for interface in eachinterface(dg, cache)
-      # Get neighboring element ids
-      left  = cache.interfaces.neighbor_ids[1, interface]
-      right = cache.interfaces.neighbor_ids[2, interface]
-
-      # Apply smoothing
-      alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
-      alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
-    end
+  
+    apply_smoothing!(u, mesh, equations, dg, alpha, alpha_tmp, cache)
+
   end
 
   return alpha
 end
 
 
+function apply_smoothing!(u::AbstractArray{<:Any, 3}, mesh::TreeMesh{1}, equations, dg::DGSEM, alpha, alpha_tmp, cache)
+
+  # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
+  # Copy alpha values such that smoothing is indpedenent of the element access order
+  alpha_tmp .= alpha
+
+  # Loop over interfaces
+  for interface in eachinterface(dg, cache)
+    # Get neighboring element ids
+    left  = cache.interfaces.neighbor_ids[1, interface]
+    right = cache.interfaces.neighbor_ids[2, interface]
+
+    # Apply smoothing
+    alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
+    alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
+  end
+  
+end
+
+
+function apply_smoothing!(u::AbstractArray{<:Any, 3}, mesh::StructuredMesh{1}, equations, dg::DGSEM, alpha, alpha_tmp, cache)
+
+  # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
+  # Copy alpha values such that smoothing is indpedenent of the element access order
+  alpha_tmp .= alpha
+
+  # Loop over elements, because there is no interface container
+  for element in eachelement(dg,cache)
```

Comment:
I think this loop will only work for fully periodic meshes, not for boundaries, or am I missing something? If true, please also check the corresponding methods for 2d and 3d

Reply:
Yes, I think this is true. Since I am not so familiar with the data structure of the interface containers, I would propose to add a note at the appropriate places in the source code and open an issue for that. What do you think @ranocha ?

Reply:
If you also add an assertion that you're only using a periodic mesh if you do that. See https://github.com/trixi-framework/Trixi.jl/blob/1c6e8997d57038557cfdc5b821d987eb14d19ac5/src/solvers/dgsem_tree/dg_1d.jl#L483

Reply:
Do you mean something like this 72a2d5a?